dev/README.md

```diff
@@ -3,101 +3,121 @@
-    #export FLEET_E2E_CLUSTER_DOWNSTREAM=k3d-upstream
-
-    # running multi-cluster tests in k3d (setup-k3ds)
-    #export FLEET_E2E_CLUSTER=k3d-upstream
-    #export FLEET_E2E_CLUSTER_DOWNSTREAM=k3d-downstream
-
-    # for running tests on darwin/arm64
-    #export GOARCH=arm64
-
-    # needed for gitrepo tests, which are currently disabled but part of the
-    # single-cluster tests
-
-    #export FORCE_GIT_SERVER_BUILD="yes" # set to an empty value to skip rebuilds
-    #export GIT_REPO_USER="git"
-    #export GIT_REPO_URL="git@github.com:yourprivate/repo.git"
-    #export GIT_REPO_HOST="github.com"
-    #export GIT_SSH_KEY="$HOME/.ssh/id_ecdsa_test"
-    #export GIT_SSH_PUBKEY="$HOME/.ssh/id_ecdsa_test.pub"
-    #export GIT_HTTP_USER="fleet-ci"
-    #export GIT_HTTP_PASSWORD="foo"
-
-    # needed for OCI tests, which are currently disabled but part of the
-    # single-cluster tests
+## Requirements
 
-    #export CI_OCI_USERNAME="fleet-ci"
-    #export CI_OCI_PASSWORD="foo"
-    #export CI_OCI_CERTS_DIR="../../FleetCI-RootCA"
-    #export HELM_PATH="/usr/bin/helm" # optional, for selecting Helm versions (see [Troubleshooting](#troubleshooting))
+- docker
+- git
+- go
+- helm
+- jq
+- k3d
+- kubectl
+- ...
 
 ## Running Tests on K3D
 
-These commands should set up k3d, and the fleet standalone images for single
-cluster tests.
-
-    export FLEET_E2E_NS=fleet-local
-    export FLEET_E2E_CLUSTER=k3d-upstream
-    export FLEET_E2E_CLUSTER_DOWNSTREAM=k3d-upstream
+These commands should set up k3d and the fleet standalone images for single
+cluster tests and run those.
 
-    dev/setup-k3d
-    dev/build-fleet
-    dev/import-images-k3d
+```bash
+dev/setup-single-cluster
+ginkgo e2e/single-cluster
+```
 
-    # needed for gitrepo tests
-    dev/import-images-tests-k3d
-    dev/create-zot-certs 'FleetCI-RootCA' # for OCI tests
-    dev/setup-fleet
-
-    # This should be needed only once
-    cd e2e/testenv/infra
-    go build -o . ./...
-    cd -
-
-    ./e2e/testenv/infra/infra setup
-
-    ginkgo e2e/single-cluster
-
-    ./e2e/testenv/infra teardown # optional
-
-Optional flags for reporting on long-running tests: `--poll-progress-after=10s --poll-progress-interval=10s`.
+Optional flags to `ginkgo` for reporting on long-running tests:
+`--poll-progress-after=10s --poll-progress-interval=10s`.
 
 For multi-cluster tests we need to configure two clusters. You also need to make
-the upstream clusters API accessible to the downstream cluster. The default
-`url` in [dev/setup-fleet-downstream] should work with most systems.
-
-    export FLEET_E2E_NS=fleet-local
-    export FLEET_E2E_CLUSTER=k3d-upstream
-    export FLEET_E2E_CLUSTER_DOWNSTREAM=k3d-downstream
+the upstream clusters API accessible to the downstream cluster. The default URL
+in `dev/setup-fleet-downstream` should work with most systems.
 
-    dev/setup-k3ds
-    dev/build-fleet
-    dev/import-images-k3d
-    dev/setup-fleet-multi-cluster
+```bash
+source dev/setup-multi-cluster
```

Comment:
Is `source` really needed? Please note that this step differs from the `single-cluster` case above

Reply:
That single and multi cluster weren't the same was a mistake. The answer is no, it is not needed for the `dev/setup-{single,multi}-cluster` scripts. They will run successfully without it. But for `ginkgo` in the following lines, it will set the environment variables correctly and `ginkgo` requires them. So, it's either `source script && ginkgo ...` or `./script && source <the-same-env-vars> && ginkgo ...`. That's why I think sourcing them should be preferred.

Reply:
Thanks for explaining!